### PR TITLE
New storage

### PR DESCRIPTION
**Pull request by sondreso**
Adds new storage :) 

(Supersedes: #834, needed an equinor branch)

### CODE REVIEW

File: ert_shared/storage/extraction_api.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+def _extract_and_dump_observations(rdb_api, blob_api):
+    facade = ERT.enkf_facade
+
+    observation_keys = [
+        facade.get_observation_key(nr) for nr, _ in enumerate(facade.get_observations())
+    ]
+
+    measured_data = MeasuredData(facade, observation_keys)
+    # TODO: Should save all info and info about deactivation
```
Comment (markusdregi) [inline ert_shared/storage/extraction_api.py:35]: This should either be fixed or persisted in an issue

File: tests/storage/test_rdb_api.py [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+        assert obs.name == "observation_one"
+
+
+def test_get_parameter(db_info):
+    populated_db, db_lookup = db_info
+    rdb_connection = connections.get_rdb_connection(populated_db)
+    with RdbApi(connection=rdb_connection) as rdb_api:
+        param = rdb_api.get_parameter(
+            name="A", group="G", realization_index=0, ensemble_name="ensemble_name"
+        )
```
Comment (ManInFez) [inline tests/storage/test_rdb_api.py:466]: Add assertion

File: ert_shared/feature_toggling.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+class _Feature:
+
+    def __init__(self, default_enabled, python3_only=False):
+        self.is_python3_only = python3_only
+        python_major_version = sys.version_info[0]
+        self.is_enabled = default_enabled and not (python_major_version < 3 and python3_only)
+        self.is_togglable = not (python_major_version < 3 and python3_only)
+
+class FeatureToggling:
+    __conf = {
```
Comment (markusdregi) [inline ert_shared/feature_toggling.py:13]: Why the double underscore?

File: ert_shared/models/multiple_data_assimilation.py [hunk 4]
```diff
@@ -56,20 +57,27 @@ def runSimulations(self, arguments):
         self.setPhaseName(phase_string, indeterminate=True)
 
         run_context = None
+        previous_ensemble_name = None
         for iteration, weight in enumerate(weights):
             run_context = self.create_context( arguments , iteration,  prior_context = run_context )
             self._simulateAndPostProcess(run_context, arguments )
+            print("Case:", ERT.enkf_facade.get_current_case_name())
```
Comment (markusdregi) [inline ert_shared/models/multiple_data_assimilation.py:64]: Please remove `print`

File: ert_shared/storage/extraction_api.py (hunk 1 above)
Comment (sondreso) [reply ert_shared/storage/extraction_api.py:35]: Already fixed, just remove TODO

File: ert_shared/storage/extraction_api.py [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+                ensemble_name=ensemble_name,
+            )
+            misfit_value = obs_vector.getTotalChi2(fs, realization_number)
+            rdb_api._add_misfit(
+                value=misfit_value, link_id=link.id, response_id=response.id
+            )
+
+
+def dump_to_new_storage(reference=None, rdb_connection=None, blob_connection=None):
+    if not FeatureToggling.is_enabled("new-storage"):
```
Comment (markusdregi) [inline ert_shared/storage/extraction_api.py:208]: This would make a nice decorator

File: ert_shared/storage/extraction_api.py [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+        return
+
+    start_time = time.time()
+    print("Starting extraction...")
```
Comment (markusdregi) [inline ert_shared/storage/extraction_api.py:212]: Please search for `print`-statements and either replace them by logging or remove them...

**Comment by mortalisk** [comment]
Test this please!

**Comment by ManInFez** [comment]
Test this please!

**Comment by ManInFez** [comment]
Test this please!

**Comment by lars-petter-hauge** [comment]
Should https://github.com/equinor/ert/issues/876 be fixed before merge?

**Comment by markusdregi** [comment]
What Lars-Petter said ⬆️ Does it make sense to merge before #876 is resolved? Does the GUI currently work?

**Comment by ManInFez** [comment]
I am resolving this at the moment, along with some bugs I detected

File: ert_shared/feature_toggling.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        self.is_togglable = not (python_major_version < 3 and python3_only)
+        self.msg = msg
+
+
+class FeatureToggling:
+    _conf = {
+        "new-storage": _Feature(
+            default_enabled=False,
+            python3_only=True,
+            msg="New storage starts a http server on your computer serving the \
```
Comment (markusdregi) [inline ert_shared/feature_toggling.py:21]: How about:
```python
msg = (
    "WARNING: The new storage solution is experimental! "
    "In particular it starts a http server on your computer serving the "
    "data from ERT and should therefore not be used on confidential data."
)
```

Comment (ManInFez) [reply ert_shared/feature_toggling.py:21]: it is logged as a waring, and thus adding warning to the message looks a bit weird. Removing the warning it looks like this when printed:
`2020-05-12 14:01:20,273 - root - WARNING - The new storage solution is experimental! In particular it starts a http server on your computer serving the data from ERT and should therefore not be used on confidential data.`

Comment (markusdregi) [reply ert_shared/feature_toggling.py:21]: My mistake 👍

**Comment by markusdregi** [comment]
Can you squash the removal of the license into the second commit?

**Comment by markusdregi** [comment]
And squash the last two commits on adding the warning as well?

**Pull request merged**